src/dataLoader/tagsDataLoader.spec.ts

```diff
@@ -0,0 +1,125 @@
+import sinon from 'sinon';
+import { batchGetTagsByItemIds } from './tagsDataLoader';
+import { writeClient } from '../database/client';
+import { SavedItemDataService, TagDataService } from '../dataService';
+import { ContextManager, IContext } from '../server/context';
+import { Tag } from '../types';
+
+describe('tags dataloader', function () {
+  const testTags: Tag[] = [
+    {
+      id: '1',
+      name: 'tag1',
+      savedItems: ['1'],
+    },
+    {
+      id: '2',
+      name: 'tag2',
+      savedItems: ['1', '2'],
+    },
+  ];
+
+  afterAll(() => {
+    sinon.restore();
+  });
+
+  beforeAll(() => {
+    sinon.restore();
+  });
+
+  it('batchGetTagsByItemIds should return empty array of Tags for non-existant Items', async () => {
+    const promiseTags = Promise.resolve(testTags);
+    const db = writeClient();
+    const context: IContext = new ContextManager({
+      request: {
+        headers: { userid: '1', apiid: '0', premium: 'true' },
+      },
+      dbClient: db,
+      eventEmitter: null,
+    });
+    const savedItemService = new SavedItemDataService(context);
+    const service = new TagDataService(context, savedItemService);
+    sinon.stub(service, 'batchGetTagsByUserItems').returns(promiseTags);
+    const tags = await batchGetTagsByItemIds(service, ['3', '5']);
+
+    // dataloader wants an array that is the same length
+    // of array of args passed in, e.g. number of Item IDs -
+    // (this is how it matches up results to batched inputs)
+    expect(tags).toHaveLength(2);
+    // there are no items with the first given tag;
+    expect(tags[0]).toHaveLength(0);
+    // there are no items with the second given tag;
+    expect(tags[1]).toHaveLength(0);
+  });
+
+  it('batchGetTagsByItemIds should return array of Tags & empty array for mix of existing & non-existing Item IDs', async () => {
+    const promiseTags = Promise.resolve(testTags);
+    const db = writeClient();
+    const context: IContext = new ContextManager({
+      request: {
+        headers: { userid: '1', apiid: '0', premium: 'true' },
+      },
+      dbClient: db,
+      eventEmitter: null,
+    });
+    const savedItemService = new SavedItemDataService(context);
+    const service = new TagDataService(context, savedItemService);
+    sinon.stub(service, 'batchGetTagsByUserItems').returns(promiseTags);
+    const tags = await batchGetTagsByItemIds(service, ['1', '3']);
+
+    // dataloader wants an array that is the same length
+    // of array of args passed in
+    // (this is how it matches up results to batched inputs)
+    expect(tags).toHaveLength(2);
+    // there are 2 items with the first given tag;
+    expect(tags[0]).toHaveLength(2);
+    expect(tags[0]).toContainEqual({
```

Comment:
don't have to change this, but for future reference this is why I installed jest-extended. you can do `expect.toHaveSameMembers` to assert that an array contains every expected element (and none others).

Reply:
yeah, didn't seem like a huge lift for this for now. but noted, thanks